### ISSUE TITLE
eddystone_scanner failing to enable scanning on some devices

### ISSUE THREAD

**Issue by p-gentili** [canonical/checkbox#2466]
### Bug Description

The scanner has been recently updated to support Extended Scanning, but the new approach seems to be failing on many devices with 
```
# Issue LE Set Extended Scan Enable to 'True' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
# Issue LE Set Extended Scan Enable to 'False' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
```

As far as I can tell, the device will not scan at all when this happens, and indeed the test is marked as failed.

### Cert-blocker Test Case

- [ ] cert-blocker

### To Reproduce

1. Launch SRU, or any test plan including the `beacon_eddystone_url_*` test.
2. The test will fail on certain machines, like https://certification.canonical.com/hardware/202411-35942

### Expected Result

1. The scanner can enable bluetooth scanning mode
2. The beacon is detected

### Actual Result

1. The scanner is NOT able to enable bluetooth scanning

### Environment

- Ubuntu Desktop 24.04
- 6.17.0-1018-oem

### Relevant log output

```shell
====================================
Attempt 1/3 (function 'beacon_scan')
====================================
# Checking the LE Extended advertising capability
Received response from controller.
Raw 8-byte LE feature mask: ff d9 1 3c ae 0 0 0
# Extended advertising support: True
# Checking the LE Extended advertising length
# Max advertising data length: 200
# Issue LE Set Extended Scan Enable to 'False' by hci command
# Issue LE Set Extended Scan Parameters by hci command
# Issue LE Set Extended Scan Enable to 'True' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
# Issue LE Set Extended Scan Enable to 'False' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Attempt 1 failed:
No EddyStone URL advertisement detected!

Waiting 5.97 seconds before retrying...

====================================
Attempt 2/3 (function 'beacon_scan')
====================================
# Checking the LE Extended advertising capability
Received response from controller.
Raw 8-byte LE feature mask: ff d9 1 3c ae 0 0 0
# Extended advertising support: True
# Checking the LE Extended advertising length
# Max advertising data length: 200
# Issue LE Set Extended Scan Enable to 'False' by hci command
# Issue LE Set Extended Scan Parameters by hci command
# Issue LE Set Extended Scan Enable to 'True' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
# Issue LE Set Extended Scan Enable to 'False' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Attempt 2 failed:
No EddyStone URL advertisement detected!

Waiting 4.62 seconds before retrying...

====================================
Attempt 3/3 (function 'beacon_scan')
====================================
# Checking the LE Extended advertising capability
Received response from controller.
Raw 8-byte LE feature mask: ff d9 1 3c ae 0 0 0
# Extended advertising support: True
# Checking the LE Extended advertising length
# Max advertising data length: 200
# Issue LE Set Extended Scan Enable to 'False' by hci command
# Issue LE Set Extended Scan Parameters by hci command
# Issue LE Set Extended Scan Enable to 'True' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
# Issue LE Set Extended Scan Enable to 'False' by hci command
Warning: HCI Command failed. Error code: 0x12, Payload: b'\x04\x0e\x04\x01B \x12'
Attempt 3 failed:
No EddyStone URL advertisement detected!

All the attempts have failed!
Traceback (most recent call last):
  File "/usr/bin/checkbox-support-eddystone_scanner", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3/dist-packages/checkbox_support/helpers/timeout.py", line 180, in _f
    return run_with_timeout(f, timeout_s, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/checkbox_support/helpers/timeout.py", line 155, in run_with_timeout
    raise exception_queue.get_nowait()
RuntimeError: No EddyStone URL advertisement detected!
```

**Comment by Hook25** [comment]
@stanley31huang please give this maximum priority as this is part of most of our SRU/SUV testplans, so it is going to be very impactful on the team until we fix it. If you need support let us know

**Comment by p-gentili** [comment]
Fix now in beta.

**Comment by p-gentili** [comment]
Re-opening because this is still a thing on UC18. 

Same machine:
- (checkbox18 Version: 7.2.0-dev22-4804) https://certification.canonical.com/hardware/201805-26255/submission/487876/
- (checkbox18 Version: 7.1.0-dev8-4626) https://certification.canonical.com/hardware/201805-26252/submission/488044/

**Comment by bladernr** [comment]
Still broken on UC18: https://certification.canonical.com/hardware/201805-26252/submission/492781/test/76077/result/59463044/

**Comment by douglasdotc** [comment]
Found this is failing on linux oem 24.04: http://10.102.156.15:8080/job/cert-oem-sru-noble-desktop-lenovo-thinkpad-t14s-gen-6-c36871/58/consoleFull
